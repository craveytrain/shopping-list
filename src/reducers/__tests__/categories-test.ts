import reducer from 'reducers/categories';
import {
  setCategories,
  addCategory,
  removeCategory,
  setCategoryName
} from 'actions/categories';
import * as deepFreeze from 'deep-freeze';

describe('categories reducer', () => {
  describe('setCategories', () => {
    it('should set categories', () => {
      const initialState = [
        {
          name: 'Dairy',
          id: 'dairy'
        }
      ];

      deepFreeze(initialState);

      const nextState = reducer(initialState, setCategories([
        {
          name: 'Meat'
        }
      ]));

      expect(nextState).toMatchSnapshot();
    });
  });

  describe('addCategory', () => {
    it('should add a category to the list', () => {
      const initialState = [
        {
          name: 'Dairy',
          id: 'dairy'
        }
      ];

      deepFreeze(initialState);

      const nextState = reducer(initialState, addCategory({
        name: 'Meat',
        id: 'meat'
      }));

      expect(nextState).toMatchSnapshot();
    });
  });

  describe('removeCategory', () => {
    it('should remove a category from the list', () => {
      const initialState = [
        {
          name: 'Dairy',
          id: 'dairy'
        },
        {
          name: 'Meat',
          id: 'meat'
        }
      ];

      deepFreeze(initialState);

      const nextState = reducer(initialState, removeCategory('dairy'));

      expect(nextState).toMatchSnapshot();
    });
  });

  describe('setCategoryName', () => {
    it('should set the name of the selected category from the list', () => {
      const initialState = [
        {
          name: 'Dairy',
          id: 'dairy'
        },
        {
          name: 'Meat',
          id: 'meat'
        }
      ];

      deepFreeze(initialState);

      const nextState = reducer(initialState, setCategoryName('dairy', 'Milk & Cheese'));

      expect(nextState).toMatchSnapshot();
    });
  });

  it('can be used with reduce', () => {
   const actionQueue = [
     setCategories([
       {
         name: 'Dairy',
         id: 'dairy'
       },
       {
         name: 'Produce',
         id: 'produce'
       }
     ]),
     addCategory({
       name: 'Meat'
     }),
     setCategoryName('dairy', 'Milk & Cheese'),
     removeCategory('produce')
   ];

   const finalState = actionQueue.reduce(reducer, []);

   expect(finalState).toMatchSnapshot();
});
