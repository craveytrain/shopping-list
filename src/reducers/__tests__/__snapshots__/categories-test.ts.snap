// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories reducer addCategory should add a category to the list 1`] = `
Array [
  Object {
    "id": "dairy",
    "name": "Dairy",
  },
  Object {
    "id": "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
    "name": "Meat",
  },
]
`;

exports[`categories reducer can be used with reduce 1`] = `
Array [
  Object {
    "id": "dairy",
    "name": "Milk & Cheese",
  },
  Object {
    "id": "c90399f41b3fd3c0af1e10c70e03eb5369d79015",
    "name": "Meat",
  },
]
`;

exports[`categories reducer nameCategory should set the name of the selected category from the list 1`] = `
Array [
  Object {
    "id": "dairy",
    "name": "Milk & Cheese",
  },
  Object {
    "id": "meat",
    "name": "Meat",
  },
]
`;

exports[`categories reducer removeCategory should remove a category from the list 1`] = `
Array [
  Object {
    "id": "meat",
    "name": "Meat",
  },
]
`;

exports[`categories reducer setCategories should set categories 1`] = `
Array [
  Object {
    "name": "Meat",
  },
]
`;
